paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt

```diff
@@ -58,6 +63,8 @@ internal abstract class BaseSheetViewModel<TransitionTargetType>(
     private val _stripeIntent = MutableLiveData<StripeIntent?>()
     internal val stripeIntent: LiveData<StripeIntent?> = _stripeIntent
 
+    internal var supportedPaymentMethods = emptyList<SupportedPaymentMethod>()
```

Comment:
is there any reason why this can't still be a `LiveData` ? this would introduce a strict order when the value is written and when the value is consumed, which is why LiveData/observer is introduced

Reply:
Previously this was a function and it was called once in BaseSheetViewModel, and again in BaseAddPaymentMethodFragment.  I tried to convert it to use the reactive model, but it started to get a bit involved.  I can follow up in another PR to make it more reactive.